haskell-tests/Types/PayloadSerializationSpec.hs

```diff
@@ -75,13 +58,45 @@ checkPayload spv e = let bs = S.runPut $ putPayload e
                       Left err -> counterexample err False
                       Right e' -> label (groupIntoSize (fromIntegral (BS.length bs))) $ e === e'
 
+modifyPayloadBitmap :: (Word16 -> Word16) -> BS.ByteString -> BS.ByteString
+modifyPayloadBitmap f bs =
+  let Right ((header, bitmap), rest) = S.runGetState ((,) <$> S.getWord8 <*> S.getWord16be) bs 0
+  in S.runPut (S.putWord8 header <> S.putWord16be (f bitmap)) `BS.append` rest
+
+genModifyPayloadBitmap :: Int -> BS.ByteString -> Gen BS.ByteString
+genModifyPayloadBitmap sizeOfBitmap payload = do
+  forcedBit <- choose (sizeOfBitmap, 15)
+  optionalBits <- mapM (\i -> elements [0, bit i]) $ delete forcedBit [sizeOfBitmap..15]
```

Comment:
```suggestion
  optionalBits <- arbitrary .&. shiftL maxBound sizeOfBitmap
```

Reply:
Yes this is a simpler approach. I was worried about endianness with shift left/right, because I thought it would shift bits to the left/right regardless of endianness. But I just realized that it is in fact a multiplication/division by a power of two, making it independent of endianness.